tests/test_config.py

```diff
@@ -155,6 +155,36 @@ def test_env_var_in_section2_dont_set(self):
         settings.GET_ENVIRON_STRICT = False
         self.assertTrue('Environment variable "SECTION2" is not set' in str(context.exception))
 
+    def test_env_var_in_include_dont_set(self):
+        settings.CONFIG_FILE = 'tests/fixtures/config_with_include_and_env_vars.yaml'
+        settings.GET_ENVIRON_STRICT = True
+        with self.assertRaises(RuntimeError) as context:
+            config.load_context_section('section-2')
+
+        settings.GET_ENVIRON_STRICT = False
+        self.assertTrue('Environment variable "INC_ENV" is not set'
+                        in str(context.exception))
+
+    def test_env_var_in_include_2_levels_dont_set(self):
+        settings.CONFIG_FILE = 'tests/fixtures/config_with_include_and_env_vars.yaml'
+        settings.GET_ENVIRON_STRICT = True
+        with self.assertRaises(RuntimeError) as context:
+            config.load_context_section('section-1')
+
+        settings.GET_ENVIRON_STRICT = False
+        self.assertTrue('Environment variable "INC_ENV" is not set'
+                        in str(context.exception))
+
+    def test_max_recursion_depth(self):
+        settings.CONFIG_FILE = 'tests/fixtures/config_with_include_and_env_vars.yaml'
+        settings.MAX_CONFIG_INCLUDE_DEPTH = 1
+        with self.assertRaises(RuntimeError) as context:
+            config.load_context_section('section-1')
+
+        settings.MAX_CONFIG_INCLUDE_DEPTH = 5
+        self.assertTrue('Max include depth for config has reached'
```

Comment:
I've seen it before, but test assertions on **exact exception text** is quite bad practice in my opinion. Gives just a bit better precision, but also makes tests less readable and changeable.

Even if we throw the same exception type everywhere (and therefore cannot distinguish them) we should assert on exception type and the fact it has been raised. That would be enough.

Not blocking, but I propose to avoid such assertions.